enum in stitching.hpp collided with X11 macro

##### System information (version)
- OpenCV => Latest
- Operating System / Platform => Ubuntu 14.04 LTS
- Compiler => GCC 4.9.3

##### Detailed description
The recent change of including stitching.hpp inside opencv.hpp is causing conflicts with X11 which results in compilation error.

##### Steps to reproduce
Write any code with the lines of 

``````
#include "X11/Xlib.h"
#include "opencv2/opencv.hpp"
``````

there will be compilation error

``````
/usr/local/include/opencv2/stitching.hpp:100:10: error: expected identifier before ‘int’
     enum Status
          ^
In file included from /usr/local/include/opencv2/opencv.hpp:86:0,
``````

the root cause is the below code in Xlib.h

``````
#define Status int
``````

It's a horrible macro but Xlib is too old to change and it's already by default in all Ubuntu distributions. Would OpenCV be willing to accommodate by changing enum Status into some other name in stitching.hpp to avoid this collision so I can use Xlib in conjunction with OpenCV for my graphics application?

--- a/modules/stitching/include/opencv2/stitching.hpp
+++ b/modules/stitching/include/opencv2/stitching.hpp
@@ -53,6 +53,12 @@
 #include "opencv2/stitching/detail/blenders.hpp"
 #include "opencv2/stitching/detail/camera.hpp"
 
+
+#if defined(Status)
+#  warning Detected X11 'Status' macro definition, it can cause build conflicts. Please, include this header before any X11 headers.
+#endif
+
+
 /**
 @defgroup stitching Images stitching
 

